### ISSUE TITLE
Misrepresentation of table values after rounding

### ISSUE THREAD

**Issue by darecstowell** [jakob/Postico#166]
I noticed that Postico was showing the wrong representation of values for random cells after I used the Postgres round(v numeric, s int) function on a whole column.
Here's the specific query - **update *schema.table* set *column* = round(cast(*column* as numeric) *100, 2)**
<img width="461" alt="2016-01-22t15-18-43" src="https://cloud.githubusercontent.com/assets/8962962/12523389/25589398-c11c-11e5-8dff-6223ceb73b59.png">
^ Incorrect representation of random value in Postico. (bottom right)^
<img width="357" alt="2016-01-22t15-40-10" src="https://cloud.githubusercontent.com/assets/8962962/12523762/773d1cd6-c11e-11e5-914b-69c59cccbadf.png">
^ Same value in psql. ^
Again, this happened for multiple values throughout the table after rounding, this is just one example.

At first, I thought I was rounding wrong, but myself and a few others checked multiple values and confirmed they were rounded correctly. 
Just thought I would share. :)

**Comment by Darecs00** [comment]
Also, I have ran basic troubleshooting, and continue to see the misrepresentation.

**Comment by jakob** [comment]
Could you check the type of the column? If the type is float8 (double precision), Postico shows 16 significant digits, but doubles only have 15.9 digits of precision, so sometimes odd things like that happen. If you need precise decimal numbers, I recommend working with the NUMERIC type.

Consider the following query:
```sql
select 0.07::float4, 0.07::float8, 0.07::numeric
```

This shows the following result:
<img width="291" alt="screen shot 2016-01-26 at 00 39 27" src="https://cloud.githubusercontent.com/assets/198305/12568330/50d19744-c3c5-11e5-8e28-16eb41a78afc.png">

Postico could just round to 15 digits when displaying numbers, then that wouldn't happen. Since a bunch of people have complained about this, I'll see what I can do.

**Comment by Darecs00** [comment]
Thanks for the response. 
Yup, they are all double precision, which made sense before the rounding. 
It's a great app. A good amount of people at my office have come over from pgAdmin. :) 
Look forward to your advancements and wish you the best!

**Comment by jakob** [comment]
I fixed this in the latest build. Postico now always uses the shortest possible decimal representation that has the same underlying floating point representation.

You can download the build here:
https://eggerapps-downloads.s3-eu-west-1.amazonaws.com/postico-1316.zip

**Comment by qwesda** [comment]
Sadly this fix messes up numeric columns:

```sql
SELECT 3.51::numeric;
```

outputs: `> 3.5100000000000002`

I have several tables with numeric columns that exhibit this behaviour.

**Comment by jakob** [comment]
@qwesda Thanks a lot for catching this.

Numeric/decimal formatting works again in the following build:
https://eggerapps-downloads.s3-eu-west-1.amazonaws.com/postico-1317.zip

**Comment by qwesda** [comment]
Seems to work ok now

**Comment by jakob** [comment]
This issue is now resolved in Postico 1.0.4
Download here: https://eggerapps.at/postico/
